[CUMULUS-1773] Consistent query params across app

dopeters opened the pull request:
**Summary:** A lot of changes for better handling of query params

Addresses [CUMULUS-1773: Keep URLs consistent and useful for dashboard users.](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1773)

## Changes

* All `index.js` files for the various pages now include the higher-order component `withQueryParams`. Query params are passed to routes that include `<List>` components via
`<Route render={(props) => <ComponentWithList {...props} queryParams={queryParams}/>`
The queryParams are used in the query that is passed to the `<List>`
   - This is the major change that allows us to go to a page with query params and have the list filtered correctly. Ex: going to http://localhost:3000/granules?status=running directly will display only the running granules
* All instances of `history.push` have been replaced with the function `historyPushWithQueryParams`. This allows for persisting the `startDateTime` and `endDateTime` params
* All instances of using the <Link> component now include passing the `search` option using `getPersistentQueryParams` function to persist the `startDateTime` and `endDateTime` params
* Various formatting fixes that made sense to do as I was changing things

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

jennyhliu commented:
Providers page doesn't have datepicker, however, the following link doesn't return any providers.  This is something different from baseline.
http://localhost:3000/providers?endDateTime=20200717172144&startDateTime=20200617172144

jennyhliu commented:
The Executions page has datepicker issue too, but it might be the baseline issue as well.  The executions older than 1 month are returned as well.
http://localhost:3000/executions?endDateTime=20200717172144&startDateTime=20200617172144

dopeters commented:
Yep so there were two things happening.

1. Executions (and Operations) were passing a `dispatch` function even though they weren't connected to the redux state. Fixed that.
2. Was not filtering out `startDateTime` and `endDateTime` from Providers page and others. Added a helper function `filterQueryParams` to do this and added it to each page

dopeters commented on app/src/js/actions/index.js:
```diff
@@ -615,14 +615,17 @@ export const clearPdrsSearch = () => ({ type: types.CLEAR_PDRS_SEARCH });
-  }
-});
+export const listProviders = (options) => {
+  console.log(options);
```
Done https://github.com/nasa/cumulus-dashboard/pull/810/commits/6a8247358e161289dfe0ae80d255e0366f6d6bae

dopeters commented on package.json:
```diff
@@ -175,6 +174,7 @@
     "@fortawesome/react-fontawesome": "^0.1.7",
     "@reduxjs/toolkit": "^1.2.5",
     "assert": "^1.4.1",
+    "babel-plugin-rewire": "^1.2.0",
```
Yep! Done https://github.com/nasa/cumulus-dashboard/pull/810/commits/185458ee04678836f59252cd15eac82512c8a825
